pet/pet.js

```diff
@@ -15,29 +16,66 @@ const addCommentForm = document.getElementById('add-comment-form');
 
 /* State */
 let error = null;
-let pet = null;
+let pet = {};
+let comments = [];
 
 /* Events */
 window.addEventListener('load', async () => {
     // > Part B:
     //   - get the id from the search params
+    const searchParams = new URLSearchParams(location.search);
+    const id = searchParams.get('id');
     //   - if no id, redirect to list (home) page
+    if (!id) {
+        location.replace('/');
+        return;
+    }
+
     //  - otherwise, get the pet by id and store the error and pet data
+    const response = await getPet(id);
+    error = response.error;
+    pet = response.data;
+    comments = pet.comments;
     //  - if error, display it
+    if (error) {
+        displayError();
+    }
```

Comment:
you'll want a redirect in here, too, because if i enter a bad id it gets an error but it doesn't redirect. and the error doesn't show up on the screen so maybe you could tweak ```displayError()```